plugins/catalog-backend/src/search/DefaultCatalogCollatorFactory.ts

```diff
@@ -67,36 +71,30 @@ export class DefaultCatalogCollatorFactory implements DocumentCollatorFactory {
     const {
       batchSize,
       discovery,
+      type,
       locationTemplate,
       filter,
       catalogClient,
       tokenManager,
+      entityTransformer,
     } = options;
 
+    this.type = type ?? 'software-catalog';
```

Comment:
Nor sure of the choice of the nullish coalescing option `??` over a simple `||`. It would make an empty string type valid which I'm not sure we want.

Reply:
removed ability to change type